packages/rich-text/src/plugins/Quote/createQuotePlugin.ts

```diff
@@ -39,5 +41,46 @@ export function createQuotePlugin(): RichTextPlugin {
         },
       },
     ],
+    withOverrides: (editor) => {
+      const { insertFragment } = editor;
+
+      editor.insertFragment = (fragment) => {
+        const startingNode = fragment.length && fragment[0];
+        const startsWithBlockquote =
+          Element.isElement(startingNode) && startingNode.type === BLOCKS.QUOTE;
+
+        let cursorEntry = getAbove(editor);
+        const cursorIsAtParagraph =
+          cursorEntry &&
+          TEXT_CONTAINERS.includes(cursorEntry?.[0]?.type) &&
```

Comment:
Are you trying to see if we are inside a text container (paragraph or heading)? If yes, then you may wan to try this:

```javascript
const containerEntry = getAbove(editor, {
  match: {
    type: TEXT_CONTAINERS,
  }
})
```

Reply:
Updated 👍🏼

Reply:
Cool. Do you want to apply the same changes to line 52?